app/src/main/java/com/nickytoolchick/agraph/MainActivity.kt

```diff
@@ -1,12 +1,34 @@
 package com.nickytoolchick.agraph
 
+import android.app.Activity
+import android.content.Intent
 import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
-import com.example.agraph.R
+import android.util.Log
+import androidx.activity.result.contract.ActivityResultContracts
+import com.nickytoolchick.agraph.data.ChartOptions
+import com.nickytoolchick.agraph.databinding.ActivityMainBinding
+import com.nickytoolchick.agraph.ui.ChartOptionsActivity
 
 class MainActivity : AppCompatActivity() {
+
+    private var chartOptions = ChartOptions()
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_main)
+        val binding = ActivityMainBinding.inflate(layoutInflater)
```

Comment:
It's very good that you use viewBinding instead of the standard unsafe `findById`